Plugins/StomtPlugin/Source/StomtPlugin/Private/StomtAPI.cpp

```diff
@@ -315,139 +279,121 @@ FString UStomtAPI::ReadLogFile(FString LogFileName)
-				{
-					this->accesstoken = Request->GetResponseObject()->GetObjectField(TEXT("meta"))->GetStringField(TEXT("accesstoken"));
-					this->SaveAccesstoken(this->accesstoken);
-				}
-			}
+			this->SendStomt(StomtToSend);
+			LogFileWasSend = false;
 		}
+	}
+}
+
+void UStomtAPI::SendEMail(FString EMail)
+{
+	UStomtRestRequest* request = this->SetupNewPostRequest();
+	request->OnRequestComplete.AddDynamic(this, &UStomtAPI::UStomtAPI::OnSendEMailResponse);
+
+	request->GetRequestObject()->SetStringField(TEXT("email"), EMail);
 
-		// Get File uid of the error log and send stomt
+	request->ProcessURL(this->GetRestURL().Append(TEXT("/authentication/subscribe")));
+}
+
+
+void UStomtAPI::OnSendEMailResponse(UStomtRestRequest * Request)
+{
+	if (Request->GetResponseCode() != 400)
+	{
 		if (Request->GetResponseObject()->HasField(TEXT("data")))
 		{
-			if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->HasField(TEXT("files")))
+			if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->HasField(TEXT("success")))
 			{
-				this->errorLog_file_uid = Request->GetResponseObject()->GetObjectField(TEXT("data"))->GetObjectField(TEXT("files"))->GetObjectField(TEXT("stomt"))->GetStringField("file_uid");
-				this->LogFileWasSend = false;
-				this->SendStomt(StomtToSend);
+				if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->GetBoolField("success"))
+				{
+					this->Config->SetSubscribed(true);
+				}
 			}
-		}
-
-		if (Request->GetResponseCode() == 403 || Request->GetResponseCode() == 419 || Request->GetResponseCode() == 413)
-		{
-			if (StomtToSend != NULL)
+			else
 			{
-				this->SendStomt(StomtToSend);
-				LogFileWasSend = false;
+				this->Config->SetSubscribed(false);
 			}
 		}
 	}
+	else
+	{
+		this->Config->SetSubscribed(true);
+	}
+}
+
+void UStomtAPI::SendLogoutRequest()
+{
+	UStomtRestRequest* request = this->SetupNewDeleteRequest();
+	request->OnRequestComplete.AddDynamic(this, &UStomtAPI::UStomtAPI::OnSendLogoutResponse);
 
-	if (EMailFlagWasSend)
+	request->ProcessURL(this->GetRestURL().Append(TEXT("/authentication/session")));
+}
+
+void UStomtAPI::OnSendLogoutResponse(UStomtRestRequest * Request)
+{
+	if (Request->GetResponseCode() != 400)
 	{
-		if (! (Request->GetResponseCode() == 400))
+		if (Request->GetResponseObject()->HasField(TEXT("data")))
 		{
-			if (Request->GetResponseObject()->HasField(TEXT("data")))
+			if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->HasField(TEXT("success")))
 			{
-				if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->HasField(TEXT("success")))
-				{
-					this->SaveFlag(TEXT("email"), true);
-				}
-				else
+				if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->GetBoolField("success"))
 				{
-					this->SaveFlag(TEXT("email"), false);
-				}
+					this->Config->SetAccessToken(TEXT(""));
+					this->Config->SetLoggedIn(false);
```

Comment:
Please also set subscribed to `false` because a new session will be created that is not connected to any subscription email address.